File: core/src/com/mygdx/game/model/obstacles/Wall.kt
package com.mygdx.game.model.obstacles

import com.badlogic.gdx.graphics.Texture
import com.badlogic.gdx.graphics.g2d.Sprite
import com.badlogic.gdx.math.Rectangle
import com.mygdx.game.settings.WALL_SPRITE_HEIGHT
import com.mygdx.game.settings.WALL_SPRITE_WIDTH

class Wall(x: Float, y: Float, texture:Texture): Sprite(texture) {

    val bounds = Rectangle(x, y, WALL_SPRITE_WIDTH, WALL_SPRITE_HEIGHT)

    init {
        setSize(WALL_SPRITE_WIDTH, WALL_SPRITE_HEIGHT)
        setPosition(x, y)
        setBounds(x, y, WALL_SPRITE_WIDTH, WALL_SPRITE_HEIGHT)
    }
}

File: server/models/obstacles/Wall.js
const Matter = require('matter-js');

const constants = require('../../settings/constants');
const BaseObject = require('../BaseObject');

class Wall extends BaseObject {
    constructor(x, y, id) {
        super(id);
        this.bounds = Matter.Bodies.rectangle(x, y, constants.WALL_SPRITE_WIDTH, constants.WALL_SPRITE_HEIGHT)
    }
}

module.exports = Wall;